### ISSUE TITLE
Investigate hive problems with enabling websocket

### ISSUE THREAD

**Issue by jochem-brouwer** [ethereumjs/ethereumjs-monorepo#2938]
In this PR https://github.com/ethereum/hive/pull/813 at hive we fix the timeout on startup since hive keeps trying to connect to RPC, but it cannot reach it. However, with the newly introduced arguments, we have to disable the websocket server, because it throws if we enable it. We should fix this.

**Comment by holgerd77** [comment]
Can you paste a concrete CL client start command which throws so to make sure I (others too of course) am getting this right?

**Comment by jochem-brouwer** [comment]
Yes, we had to disable websocket on hive in order to make hive / our client not hang on tests, see;

https://github.com/ethereum/hive/blob/44510de419e0f58dba60a39d27636362f731a184/clients/ethereumjs/ethereumjs.sh#L51

Note that we launch `--ws false`. If you remove this (so just use `--ws`, remove the `false`) then the client will start but immediately complain about reused ports. (If you want I can post the error here as well, but this should turn up if you remove the `false` from the ws and then just run any hive test, e.g. ./hive --sim "ethereum/engine" --client ethereumjs --sim.limit="engine-cancun/Request Blob Pooled Transactions")

**Comment by holgerd77** [comment]
Ok, so, to clarify further: so this is only a specific Hive thing and we do not have a very generic websocket activation bug in the client right now? Is this correct?

**Comment by jochem-brouwer** [comment]
Yes. Try this locally (after rebuilding monorepo)

`node ./dist/bin/cli.js --rpc --rpcEngine --rpcAddr 0.0.0.0 --rpcEngineAddr 0.0.0.0 --rpcEnginePort 0.0.0.0 --rpcEnginePort 8551 --ws --dataDir=./data`

Crashes after a few seconds:

```
08-14|14:34:30] INFO Started JSON RPC Server address=ws://localhost:8545 namespaces=eth,web3,net,admin,txpool,debug 
[08-14|14:34:30] ERROR Error starting client The argument 'options' must have the property "port" or "path". Received [ NaN, 8551 ]
node:events:491
      throw er; // Unhandled 'error' event
      ^

Error: listen EADDRINUSE: address already in use 0.0.0.0:8545
    at Server.setupListenHandle [as _listen2] (node:net:1740:16)
    at listenInCluster (node:net:1788:12)
    at doListen (node:net:1937:7)
    at process.processTicksAndRejections (node:internal/process/task_queues:83:21)
Emitted 'error' event on Server instance at:
    at emitErrorNT (node:net:1767:8)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  code: 'EADDRINUSE',
  errno: -98,
  syscall: 'listen',
  address: '0.0.0.0',
  port: 8545
}

Node.js v18.16.0
```

**Comment by holgerd77** [comment]
Ok, but then that's some - somewhat - serious bug in the client we should address soonish and eventually even do a hotfix release? 🤔

**Comment by holgerd77** [comment]
Guess that's likely some follow-up task for @acolytec3, right? Since he did the port forwarding work (correct me if I am wrong)?

**Comment by jochem-brouwer** [comment]
I'll do some more debugging, the title + description might be a bit misleading, upon a few mins of testing it seems to have to do with `rpcEngineAddr` and/or `rpcAddr` set to `0.0.0.0` combined with `--ws`. So it seems rather specific

**Comment by holgerd77** [comment]
Closed by #3083
